lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -166,6 +166,10 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
     GenServer.cast(pid, {:stop_smart_cell, ref})
   end
 
+  def put_system_envs(pid, secrets) do
+    GenServer.cast(pid, {:put_system_envs, secrets})
```

Comment:
I think we can call put_env directly here, instead of a cast, as the session already serializes those operations?

Reply:
@josevalim here we would set it in the Livebook node rather than the runtime node :D